File: ResourceManager/ViewModels/Users/UserRolesPopupViewModel.cs
using NetCore.Mvvm.Commands;
using NetCore.Mvvm.Controls;
using NetCore.Mvvm.ViewModels;
using NetCore.Security;
using ResourceManager.Models;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Input;

namespace ResourceManager.ViewModels
{
    public class UserRolesPopupViewModel: PopupViewModelBase
    {
        public UserRolesPopupViewModel()
        {
            AddCommand = new RelayCommand<Role>(AddRole, x => SelectedRole != null);
            RemoveCommand = new RelayCommand<Role>(RemoveRole, x=>SelectedUserRole != null);
            SaveCommand = new RelayCommand(Save);
        }

        private void Save()
        {
            Result = PopupResult.OK;
            OnClose();
        }

        private void RemoveRole(Role role)
        {
            UserRoles = UserRoles.Where(x => x.ID != role.ID).ToList();
            Roles = Roles.Append(role).ToList();
        }

        private void AddRole(Role role)
        {
            UserRoles = UserRoles.Append(role).ToList();
            Roles = Roles.Where(x => x.ID != role.ID).ToList();
        }
        public IList<Role> UserRoles
        {
            get => GetProperty<IList<Role>>();
            set => SetProperty(value);
        }
        public Role SelectedUserRole
        {
            get => GetProperty<Role>();
            set => SetProperty(value);
        }
        public IList<Role> Roles
        {
            get => GetProperty<IList<Role>>();
            set => SetProperty(value);
        }
        public Role SelectedRole
        {
            get => GetProperty<Role>();
            set => SetProperty(value);
        }
        public bool CanEditUser
        {
            get
            {
                return AuthManager.User.IsInPermission("a_edit_user");
            }
        }
        public ICommand AddCommand { get; }
        public ICommand RemoveCommand { get; }
        public ICommand SaveCommand { get; }
    }
}

File: ResourceManager/ViewModels/Users/UserRolesPopupViewModel.vb
Imports VBNetCore.Mvvm.Commands
Imports VBNetCore.Mvvm.Controls
Imports VBNetCore.Mvvm.ViewModels
Imports VBNetCore.Security
Imports ResourceManager.Models
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Input

Namespace ResourceManager.ViewModels
    Public Class UserRolesPopupViewModel
        Inherits PopupViewModelBase
        Public Sub New()
            AddCommand = New RelayCommand(Of Role)(AddressOf AddRole, Function(x) SelectedRole IsNot Nothing)
            RemoveCommand = New RelayCommand(Of Role)(AddressOf RemoveRole, Function(x) SelectedUserRole IsNot Nothing)
            SaveCommand = New RelayCommand(AddressOf Save)
        End Sub

        Private Sub Save()
            Result = PopupResult.OK
            OnClose()
        End Sub

        Private Sub RemoveRole(ByVal role As Role)
            UserRoles = UserRoles.Where(Function(x) x.ID <> role.ID).ToList()
            Roles = Roles.Append(role).ToList()
        End Sub

        Private Sub AddRole(ByVal role As Role)
            UserRoles = UserRoles.Append(role).ToList()
            Roles = Roles.Where(Function(x) x.ID <> role.ID).ToList()
        End Sub
        Public Property UserRoles As IList(Of Role)
            Get
                Return GetProperty(Of IList(Of Role))()
            End Get
            Set(ByVal value As IList(Of Role))
                SetProperty(value)
            End Set
        End Property
        Public Property SelectedUserRole As Role
            Get
                Return GetProperty(Of Role)()
            End Get
            Set(ByVal value As Role)
                SetProperty(value)
            End Set
        End Property
        Public Property Roles As IList(Of Role)
            Get
                Return GetProperty(Of IList(Of Role))()
            End Get
            Set(ByVal value As IList(Of Role))
                SetProperty(value)
            End Set
        End Property
        Public Property SelectedRole As Role
            Get
                Return GetProperty(Of Role)()
            End Get
            Set(ByVal value As Role)
                SetProperty(value)
            End Set
        End Property
        Public ReadOnly Property CanEditUser As Boolean
            Get
                Return AuthManager.User.IsInPermission("a_edit_user")
            End Get
        End Property
        Public ReadOnly Property AddCommand As ICommand
        Public ReadOnly Property RemoveCommand As ICommand
        Public ReadOnly Property SaveCommand As ICommand
    End Class
End Namespace
